src/sys/mman.rs

```diff
@@ -426,8 +426,7 @@ pub unsafe fn mmap<F: AsFd>(
     f: Option<F>,
```

Comment:
Since we now have a dedicated interface for anonymous `mmap`, seems like we should also update the `f` argument of `mmap()` to `f: F`, and add the following document:

> For anonymous mappings (MAP_ANON/MAP_ANONYMOUS), see [mmap_anonymous](https://docs.rs/rustix/latest/rustix/mm/fn.mmap_anonymous.html).

Reply:
BTW, the document of this function:

> **allocate memory**, or map files or devices into memory

Should be updated as well since it is not for memory allocations anymore?

Reply:
It can still be used to allocate memory, e.g. for shared memory mappings based on descriptors received from `shm_open`.

Reply:
> It can still be used to allocate memory, e.g. for shared memory mappings based on descriptors received from shm_open.

Thanks for showing me this! :)